ethereum/block/wait_for_receipt.go

```diff
@@ -29,6 +30,10 @@ var (
 // we return an error describing why.
 func WaitForReceipt(ctx context.Context, ec *ethclient.Client, txHash ethcommon.Hash) (*ethtypes.Receipt, error) {
 	for i := 0; i < maxRetries; i++ {
+		if ctx.Err() != nil {
+			return nil, ctx.Err()
+		}
```

Comment:
Is that an acceptable way of checking the context status?  I've been writing this, which consumes more lines, but only because that's the way I've seen other code doing the check. I don't know what is considered best practice.
```go
	select {
	case <-ctx.Done():
		return nil, ctx.Err()
	default:
	}

```

Reply:
functionally these are the same, as cancelling a context sets the error and also closes the `done` channel: https://cs.opensource.google/go/go/+/refs/tags/go1.19.3:src/context/context.go;l=397

not sure which is preferred, imo they're the same so either is ok. if it's not in a select statement I don't really see the need to use `ctx.Done`